src/Agent.Worker/Program.cs

```diff
@@ -5,17 +5,24 @@
 using System;
 using System.Globalization;
 using System.Threading.Tasks;
+using Agent.Sdk;
 
 namespace Microsoft.VisualStudio.Services.Agent.Worker
 {
     public static class Program
     {
         public static int Main(string[] args)
         {
-            // We can't use the new SocketsHttpHandler for now for both Windows and Linux
+            // In .NET Core 2.1, we couldn't use the new SocketsHttpHandler for now for both Windows and Linux
```

Comment:
It does feel like this block of code is repeated a bunch. Does it make sense to pull it into a PlatformUtil.ProgramInit() static method?

Reply:
I moved the ugly logic into `PlatformUtil`. The callsites still manipulate `AppContext` themselves.